kimchi/src/circuits/constraints.rs

```diff
@@ -200,15 +200,78 @@ pub struct ConstraintSystem<F: PrimeField> {
     pub endo: F,
     /// lookup constraint system
     #[serde(bound = "LookupConstraintSystem<F>: Serialize + DeserializeOwned")]
-    pub lookup_constraint_system: Option<LookupConstraintSystem<F>>,
+    pub lookup_constraint_system: Arc<LookupConstraintSystemCache<F>>,
     /// precomputes
     #[serde(skip)]
-    precomputations: OnceCell<Arc<DomainConstantEvaluations<F>>>,
+    precomputations: Arc<LazyCache<Arc<DomainConstantEvaluations<F>>>>,
 
     /// Disable gates checks (for testing; only enables with development builds)
     pub disable_gates_checks: bool,
 }
 
+pub(crate) type LookupConstraintSystemCache<F> =
```

Comment:
Why do you have the `deserialize` separated?

Reply:
It was @richardpringle suggestion to avoid implementing a `Default` for `LazyCache` 😅 (required for serde)

Reply:
Makes sense.